src/sequence_statistics.py

```diff
@@ -0,0 +1,229 @@
+import requests
+import random
+import json
+
+
+# API_URL = 'http://0.0.0.0:5000'
+API_URL = 'http://127.0.0.1:5001'
+
+
+# -------------- Fonctions pour calculer les stats sur les syllabes ------------------------- #
+
+
+# Ici, pour un mot donné, on incrémente les statistiques de ses syllabes adjacentes
+def get_statistics(language,word):
+    i = 0
+    length = len(word['syllables_ipa'])
+    for syllable in word['syllables_ipa']:
+
+        # récupération de la syllabe dans la BD ou création d'une nouvelle
+        syll = requests.get(f'{API_URL}/{language}_syllables/{syllable}').json()['result']
+        new = False
+        if type(syll) is str:
+            new = True
+            try :
+                ortho = word['syllables'][i]
+            except (KeyError,IndexError):
+                ortho = ""
+            syll = {
+                'language' : language,
+                'ipa_syllable' : syllable,
+                'orthographical_syllable' : ortho,
+                'preceding_ipa_syllable' : {},
+                'following_ipa_syllable' : {}
+            }
+
+        # ajout de la syllabe précédente dans les statistiques
+        if i > 0:
+            previous = word[i-1]
+        else:
+            previous = " "
+        try:
+            syll['preceding_ipa_syllable'][previous] += 1
+        except KeyError:
+            syll['preceding_ipa_syllable'][previous] = 1
+        except TypeError:
+            syll['preceding_ipa_syllable'] = {previous : 1}
+
+        # ajout de la syllabe suivante dans les statistiques
+        if i < length - 1:
+            next = word[i+1]
+        else:
+            next = " "
+        try:
+            syll['following_ipa_syllable'][next] += 1
+        except KeyError:
+            syll['following_ipa_syllable'][next] = 1
+        except TypeError:
+            syll['following_ipa_syllable'] = {next : 1}
+
+        # mettre à jour dans la DB
+        if not new:
+            requests.patch(f"{API_URL}/{language}_syllables/{syllable}", headers={'Content-Type': 'application/json'},
+                           data=json.dumps(syll))
+            print('syllabe modifiée : ',syll)
+        else:
+            requests.post(f"{API_URL}/{language}_syllables/", headers={'Content-Type': 'application/json'},
+                          data=json.dumps(syll))
+            print('syllabe ajoutée : ', syll)
+
+        i += 1
+
+    return 'Done'
+
+
+# Cette fonction permet de calculer toutes les stats sur les syllabes adjacentes dans une langue donnée à l'aide
+# de la fonction précédente
+def stats(language):
+    erase_stats(language)
+    i = 0
+    for word in requests.get(f'{API_URL}/{language}/').json()["result"]:
+        if i == 40 :
```

Comment:
this 40 is quite weird IMO. it should be at best an argument of the function.

also instead of adding one, you can do:
`for i, word in enumerate(requests.get(...)):`